File: ios/Animations/Animations/Glasses/AnimationsScreen.swift
//Copyright (c) 2020 Everysight LTD

import Foundation
import NativeEvsKit

class AnimationsScreen:Screen {
    private var cnt = 0
    private let text = Text()
    private let simpleCtrl1 = SimpleControl()
    private let simpleCtrl2 = SimpleControl()
    override func onCreate() {
        
        text.setText(text: "Animations")
            .setResource(font: Font(stockFont: .medium))
            .setTextAlign(align: Align.center)
            .setX(x: getWidth()/2)
            .setY(y: getHeight()/2)
            .setForegroundColor(color: EvsColor.green.rgba)
        
        add(uiElement: text)
        
        simpleCtrl1.setWidth(width: 30.0)
            .setHeight(height: 30.0)
            .setX(x: 30.0)
            .setY(y: 30.0)
        
        add(uiElement: simpleCtrl1)
        
        simpleCtrl2
            .setWidth(width: 70.0)
            .setHeight(height: 30.0)
            .setX(x: 230.0)
            .setY(y: 130.0)
        
        add(uiElement: simpleCtrl2)
    }
    
    override func onResume() {
        super.onResume()
        //NOTE: text and image ui elements can be animated only with translation (no scale, no rotate)
        text.animator().duration(ms: 3000).translateTo(x: text.getX(), y: 0).reverse().start()
        
        simpleCtrl1.animator()
            .scaleX(scaleFactor: 2.0)
            .scaleY(scaleFactor: 0.5)
            .duration(ms: 5000)
            .translateTo(x: 130.0,y: 30.0)
            .bounce()
            .start()
        
        simpleCtrl2.animator()
            .scale(scaleFactor: 1.5)
            .rotate(angleDegrees: 45.0)
            .repeat()
            .duration(ms: 1000)
            .register(listener: self)
            .start()
    }
}

extension AnimationsScreen : AnimatorIEvsAnimationListener {
    func animationEnded(uiElement: UIElement) {
        cnt+=1
    }
    
    func animationLooped(uiElement: UIElement) {
        cnt+=1
    }
    
    func animationStarted(uiElement: UIElement) {
        
    }
    
    func animationStep(uiElement: UIElement, progress: Double) {
        self.text.setText(text: "Loop \(cnt) \(round(progress * 100))%")
    }
}

File: android/Animations/src/main/java/com/everysight/sdk/samples/animations/AnimationsScreen.kt
package com.everysight.sdk.samples.animations

import UIKit.app.Animator
import UIKit.app.Screen
import UIKit.app.data.Align
import UIKit.app.data.EvsColor
import UIKit.app.resources.Font
import UIKit.widgets.Text
import UIKit.widgets.UIElement
import kotlin.math.round

class AnimationsScreen:Screen() {

    private val text = Text()
    private val simpleCtrl1 = SimpleControl()
    private val simpleCtrl2 = SimpleControl()

    override fun onCreate() {
        text.setText("Animations").setResource(Font.StockFont.Medium).setTextAlign(Align.center)
        text.setX(getWidth()/2).setY(getHeight()/2).setForegroundColor(EvsColor.Green.rgba)
        add(text)

        simpleCtrl1.setWidth(30f).setHeight(30f).setX(30f).setY(30f)
        simpleCtrl2.setWidth(70f).setHeight(30f).setX(230f).setY(130f)

        add(simpleCtrl1)
        add(simpleCtrl2)
    }

    override fun onResume() {
        super.onResume()
        //NOTE: text and image ui elements can be animated only with translation (no scale, no rotate)
        text.animator().duration(3000).translateTo(10f,10f).reverse().start()
        simpleCtrl1.animator()
            .scaleX(2f).scaleY(0.5f)
            .duration(5000)
            .translateTo(130f,30f)
            .bounce()
            .start()

        simpleCtrl2.animator()
            .scale(1.5f)
            .rotate(45f)
            .repeat()
            .duration(1000)
            .register(object : Animator.IEvsAnimationListener{
                var cnt = 0
                override fun animationEnded(uiElement: UIElement) {
                }

                override fun animationLooped(uiElement: UIElement) {
                    cnt++
                }

                override fun animationStarted(uiElement: UIElement) {
                }

                override fun animationStep(uiElement: UIElement, progress: Double) {
                    text.setText("Loop $cnt ${round(progress * 100)}%")
                }

            })
            .start()
    }
}